### PR TITLE
gnb and ue put added

### PR DESCRIPTION
**Pull request by DariushMoh**
# Description

gnb and ue put added

# Reviewers list 

- [ ] @edvgui   

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Correct, in line with design
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented (classes, methods and function docs, document important assumptions, link to reference materials)
- [ ] Type annotations
- [ ] Issue is in the review column
- [x] MR is assigned to first reviewer

### CODE REVIEW

File: src/nfv_test_api/v2/controllers/gnodeb.py [hunk 1]
```diff
@@ -127,6 +128,33 @@ def get(self, nci: str):
 
         return self.gnb_service.get_one(nci).json_dict(exclude_none=True), HTTPStatus.OK
 
+    @namespace.expect(gnodeb_update_model)
+    @namespace.response(
+        HTTPStatus.OK.value, "The gNodeB config has been updated", gnodeb_model
+    )
+    @namespace.response(
+        HTTPStatus.NOT_FOUND.value, "Couldn't find any gNodeB with given nci"
+    )
```
Comment (edvgui) [inline src/nfv_test_api/v2/controllers/gnodeb.py:137]: The whole point of put is that you can use it whether something is there or not.  It is idempotent, you just the resource there, regardless of what might or might not be there

File: src/nfv_test_api/v2/controllers/gnodeb.py [hunk 2]
```diff
@@ -127,6 +128,33 @@ def get(self, nci: str):
 
         return self.gnb_service.get_one(nci).json_dict(exclude_none=True), HTTPStatus.OK
 
+    @namespace.expect(gnodeb_update_model)
```
Comment (edvgui) [inline src/nfv_test_api/v2/controllers/gnodeb.py:131]: I think you should take the create model as input here

File: src/nfv_test_api/v2/controllers/gnodeb.py (hunk 1 above)
Comment (DariushMoh) [reply src/nfv_test_api/v2/controllers/gnodeb.py:137]: Okay should I pass the nci as argument as well ? Or do I move the put in the same class as the post?

File: src/nfv_test_api/v2/services/gnodeb.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def kill(self, identifier: str) -> None:
             raise NotFound(f"No process running for gNodeB with nci {identifier}")
 
 
-class GNodeBService(BaseService[GNodeB, GNodeBCreate, GNodeBUpdate]):
+class GNodeBService(BaseService[GNodeB, GNodeBCreate, GNodeBCreate]):
```
Comment (edvgui) [inline src/nfv_test_api/v2/services/gnodeb.py:81]: Why changing this? We might want to support update at some point, put is just not it

Comment (DariushMoh) [reply src/nfv_test_api/v2/services/gnodeb.py:81]: Okay so the update is PATCH ?

File: src/nfv_test_api/v2/services/gnodeb.py [hunk 4]
```diff
@@ -161,7 +161,22 @@ def create(self, o: GNodeBCreate) -> GNodeB:
         existing_gnb = self.get_one_or_default(o.nci)
         if not existing_gnb:
             raise RuntimeError(
-                "The gNodeB config should have been created but can not be found"
+                "Unexpected error: the created gNodeB config can not be found."
+            )
+
+        return existing_gnb
+
+    def put(self, o: GNodeBCreate) -> GNodeB:
```
Comment (edvgui) [inline src/nfv_test_api/v2/services/gnodeb.py:169]: Looks good, you could do an easy refactor by calling put in create

File: src/nfv_test_api/v2/services/gnodeb.py (hunk 3 above)
Comment (edvgui) [reply src/nfv_test_api/v2/services/gnodeb.py]: To keep things simple, yes

**Review by edvgui — APPROVED** [review]

**Pull request merged**